### ISSUE TITLE
[prefer-readonly-parameter-types] Using prefer-readonly-parameter-types rule with lib.dom

### ISSUE THREAD

**Issue by Denis-Afanasyev** [typescript-eslint/typescript-eslint#1727]
I can't figure out how to use `prefer-readonly-parameter-types` rule on interop with DOM (lib.dom typings).

Previously i using `@typescript-eslint/eslint-plugin: ^2.20.0` and have code like this:
```
class SomeClass {
...
    public someFunc(element: HTMLElement): void {
        ...
    }
}
```

After update to `2.23.0` i got a eslint error
`Parameter should be a read only typeeslint(@typescript-eslint/prefer-readonly-parameter-types)`
 This rule require something like
```
class SomeClass {
...
    public someFunc(element: Readonly<HTMLElement>): void {
        ...
    }
}
```

After that i got next error on the same place `Make all properties in T readonly`.
It means that i should modify external library's code to fix that error, but it's impossible

I think that lib.dom (HTMLElement, etc) just 1 out of many errors from this rule on project-library interop layer.

This error caused by https://github.com/typescript-eslint/typescript-eslint/pull/1513
So I suggest to switch rule level from `error` to `warn` in `packages/eslint-plugin/src/configs/all.json`

**Comment by bradzacher** [comment]
See https://github.com/typescript-eslint/typescript-eslint/issues/1668
This is intentional - the rule enforces that objects are _deeply_ readonly, not just surface readonly.

Ie `HTMLElement` et al are not deeply readonly, even with `Readonly<>`.
As this code is valid:
```ts
declare const x: Readonly<HTMLElement>;
x.classList = 1; // error assigning to readonly property
x.classList.foo = 1; // no readonly error, as `classList` is not readonly.
```

This is listed in the rule's docs:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-readonly-parameter-types.md#rule-details

> A type is considered readonly if:
> ...
> - it is an object type whose properties are all marked as readonly, and whose values are all considered readonly.
